CRM/RcBase/Test/BaseHeadlessTestCase.php

```diff
@@ -0,0 +1,214 @@
+     *
+     * @throws CRM_Extension_Exception_ParseException
+     */
+    public static function setUpBeforeClass(): void
+    {
+        // Resets DB
+        Test::headless()
+            ->installMe(__DIR__)
+            ->apply(true);
+    }
+
+    /**
+     * The setupHeadless function runs at the start of each test case, right before
+     * the headless environment reboots.
+     *
+     * It should perform any necessary steps required for putting the database
+     * in a consistent baseline -- such as loading schema and extensions.
+     *
+     * The utility `\Civi\Test::headless()` provides a number of helper functions
+     * for managing this setup, and it includes optimizations to avoid redundant
+     * setup work.
+     *
+     * @see \Civi\Test
+     */
+    public function setUpHeadless()
+    {
+        return Test::headless()
+            ->installMe(__DIR__)
+            ->apply();
+    }
+
+    /**
+     * Setup method run before each test
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    /**
+     * Get next ID in sequence (auto-increment)
+     *
+     * @return string Next ID
+     */
+    protected static function getNextExternalID(): string
+    {
+        self::$externalID++;
+
+        return (string)self::$externalID;
+    }
+
+    /**
+     * Get next contact sequence number (auto-increment)
+     *
+     * @return int Next ID
+     */
+    protected static function getNextContactSequence(): int
+    {
+        self::$contactSequence++;
+
+        return self::$contactSequence;
+    }
+
+    /**
+     * Sample contact parameters, next in the sequence
+     *
+     * @return array Contact parameters
+     */
+    protected function nextSampleIndividual(): array
+    {
+        // Assemble Contact data
+        $contact = $this->sampleContact('Individual', self::getNextContactSequence());
+        $contact['external_identifier'] = self::getNextExternalID();
+        // Remove unnecessary fields
+        unset($contact['prefix_id']);
+        unset($contact['suffix_id']);
+        // Remove email
+        unset($contact['email']);
+
+        return $contact;
+    }
+
+    /**
+     * Call cv api4 with get action
+     *
+     * @param string $entity Entity to work on (Contact, Email, etc.)
+     * @param array $select Fields to return
+     *   Example:
+     *   $select = ['contact_type', 'first_name', 'external_identifier']
+     * @param array $where Where conditions to filter results (if more given they are joined by AND)
+     *   Example:
+     *   $where = [
+     *     'contact_type=Individual',
+     *     'first_name like "Adams%",
+     *   ]
+     *
+     * @return array Results
+     *
+     * @throws CRM_Core_Exception
+     */
+    protected function cvApi4Get(string $entity, array $select = [], array $where = []): array
+    {
+        if (empty($entity)) {
+            throw new CRM_Core_Exception('Missing entity name');
```

Comment:
I think the test case could be failed here.
```
$this->fail("Missing entity name");
```

Reply:
updated to `fail()`